src/components/producerItems/components/ProductEditModal.js

```diff
@@ -0,0 +1,114 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Form, Input, Modal } from 'antd';
+
+import {
+    DEFAULT_FIRST_FIELD,
+    EDIT_FORM_TITLE,
+    EDITABLE_FIELDS,
+    FORM_ITEM_LAYOUT,
+    SAVE_PRODUCT_DETAILS
+} from '../constants';
+
+const FormItem = Form.Item;
+
+/**
+ * Helper function used to generate input fields for `ProductEditForm`.
+ * It makes sure that the title/name field is at the beginning of the edit form
+ *
+ * @function
+
+ * @param {Function} decorator - sets the appropriate rules and defaults on the
+ * input field
+ * @param {Object} productToEdit - the product details to edit
+ * @return {Array} - a list of field inputs
+ */
+function generateProductEditForm(decorator, productToEdit) {
```

Comment:
what is happening in line 26 to 45

Reply:
I am creating the form inputs use the array method `[reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce). If you ask more specific questions I can answer that